types/babel-plugin-tester/index.d.ts

```diff
@@ -0,0 +1,293 @@
+     * to something you hard-code, then specify snapshot: true. This will take a
+     * snapshot with both the source code and the output, making the snapshot
+     * easier to understand.
+     */
+    snapshot?: boolean;
+
+    /**
+     * If a particular test case should be throwing an error, you can that using
+     * one of the following:
+     *
+     * ```ts
+     * {
+     *  // ...
+     *   error: true,
+     *   error: 'should have this exact message',
+     *   error: /should pass this regex/,
+     *   error: SyntaxError, // should be instance of this constructor
+     *   error: err => {
+     *     if (err instanceof SyntaxError && /message/.test(err.message)) {
+     *       return true; // test will fail if function doesn't return `true`
+     *     }
+     *   },
+     * }
+     * ```
+     */
+    error?: boolean | string | RegExp | Error | ((error: unknown) => boolean);
+
+    /**
+     * If you need something set up before a particular test is run, you can do
+     * this with setup. This function will be run before the test runs. It can
+     * return a function which will be treated as a teardown function. It can
+     * also return a promise. If that promise resolves to a function, that will
+     * be treated as a teardown function.
+     */
+    setup?: () =>
+        | void
+        | NonNullable<TestObject['teardown']>
+        | Promise<void>
+        | Promise<NonNullable<TestObject['teardown']>>;
+
+    /**
+     * If you set up some state, it's quite possible you want to tear it down.
+     * You can either define this as its own property, or you can return it from
+     * the setup function. This can likewise return a promise if it's
+     * asynchronous.
+     */
+    teardown?: () => void | Promise<void>;
+
+    /**
+     * This defaults to a function which formats your code output with prettier.
+     * If you have prettier configured, then it will use your configuration. If
+     * you don't then it will be default configuration.
+     *
+     * If you'd like to specify your own, then feel free to do so. Here's the
+     * API:
+     *
+     * ```ts
+     * function customFormatter(code, { filename }) {
+     *   return formattedCode;
+     * }
+     * ```
+     *
+     * The use case for this originally was for testing codemods and formatting
+     * their result with prettier-eslint.
+     */
+    formatResult?: (code: string, options: { filename: string }) => string;
+
+    /**
+     * To use `babel.config.js` instead of `.babelrc`, set babelOptions to the
+     * config object:
+     *
+     * ```ts
+     * pluginTester({
+     *   plugin: yourPlugin,
+     *   ...
+     *   babelOptions: require('./babel.config.js'),
+     *   ...
+     *   tests: {
+     *     // your test objects
+     *   },
+     * });
+     * ```
+     *
+     */
+    babelOptions?: Babel.TransformOptions;
+}
+
+type Omit<T, K extends keyof any> = Pick<T, Exclude<keyof T, K>>;
```

Comment:
```suggestion
export type Omit<T, K extends keyof any> = Pick<T, Exclude<keyof T, K>>;
```
that's ok, and will remove that hack with `export {}`

Reply:
The problem with that is that it overloads the `Omit` import for pre ts@3.5 projects. Auto completion for `Omit` or similar popular utilities shows multiple packages and it's often difficult to know which project the type is going to be imported from.

It's just something I've noticed.